java/spi.java.hints/src/org/netbeans/modules/java/hints/spiimpl/JackpotTrees.java

```diff
@@ -61,7 +60,36 @@
 public class JackpotTrees {
 
     private static final Map<Class<?>, Class<?>> baseClass2Impl = new HashMap<>();
-    public static <T> T createInstance(Context ctx, Class<T> clazz, Name ident, JCIdent jcIdent, Class<?>[] requiredConstructor, Object[] params) {
+
+    // JDK 8-11
+    public static JCCase createJCCase(Name ident, JCIdent jcIdent, com.sun.tools.javac.util.List<?> stats) {
+        return createInstance(JCCase.class, ident, jcIdent,
+                new Class<?>[] {JCExpression.class, com.sun.tools.javac.util.List.class},
+                new Object[] {jcIdent, stats});
+    }
+
+    // JDK 12-17+
+    public static JCCase createJCCase(Name ident, JCIdent jcIdent, com.sun.tools.javac.util.List<?> labels, com.sun.tools.javac.util.List<?> stats, JCTree body) throws ReflectiveOperationException {
+        
+        // WARNING: CaseKind is deprecated for removal
+        @SuppressWarnings("rawtypes")
+        Class caseKind = Class.forName("com.sun.source.tree.CaseTree$CaseKind", false, JCCase.class.getClassLoader());
+        @SuppressWarnings("unchecked")
+        Object statement = Enum.valueOf(caseKind, "STATEMENT");
+
+        return createInstance(JCCase.class, ident, jcIdent,
+                    new Class<?>[] {caseKind, com.sun.tools.javac.util.List.class, com.sun.tools.javac.util.List.class, JCTree.class},
+                    new Object[] {statement, labels, stats, body});
+    }
```

Comment:
@jlahoda thanks for review. CaseKind is deprecated for removal but there is no other constructor on JDK 11+. Shouldn't there be one by now to simplify migration? I kinda want to deprecate this whole method so that we don't forget about that ticking bomb.

Reply:
disregard the deprecation comment, i didn't end up deprecating it. But if a non-deprecated constructor could be added to the compiler api - that would be nice for sure.
https://github.com/openjdk/jdk/blob/4f9d6b7d68312860d3e184a1f33687e1693d7d87/src/jdk.compiler/share/classes/com/sun/tools/javac/tree/JCTree.java#L1325

Reply:
Sorry, I forgot to write about `CaseKind` - it actually is not deprecated anymore. It was deprecated for removal when it was part of a preview feature, but switch expressions are a stable feature now, and `CaseKind` is an ordinary enum. (For the record, deprecating new API elements for removal is an idiom that is no longer used - the new idiom is annotating the element with `PreviewFeature`.)

Reply:
ah I see. That makes sense and also explains why there is no second constructor. I will remove the comment to avoid confusion and make the method parameters a bit nicer (case kind value can be passed as String, now that i know its here to stay).